### PR TITLE
Sharded serial leveldb

### PR DESCRIPTION
**Pull request by ssd04**

### CODE REVIEW

File: leveldb/shardedLeveldbSerial_test.go [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+	require.Nil(t, err)
+
+	err = db2.DestroyClosed()
+	require.Nil(t, err)
+
+}
+
+// ---------- Benchmarks ----------
+
+func BenchmarkPersisterOneKey1mil(b *testing.B) {
```
Comment (AdoAdoAdo) [inline leveldb/shardedLeveldbSerial_test.go]: can you move the benchmarks into a different file?
e.g shardedLeveldbSerial_bench_test.go

also I would remove the *OneKey* benchmarks as these take too long and the benchmark timeouts.

File: leveldb/shardedLeveldbSerial_bench_test.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+	b.Run("persister", func(b *testing.B) {
+		benchmarkPersister(b, keys, persisterPath, singleID)
+	})
+
+	b.Run("sharded persister", func(b *testing.B) {
+		benchmarkPersister(b, keys, shardedPersisterPath, shardedID)
+	})
+}
+
+type persister interface {
```
Comment (AdoAdoAdo) [inline leveldb/shardedLeveldbSerial_bench_test.go]: not used, can be removed

File: leveldb/shardedLeveldbSerial_bench_test.go [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+
+	err = db.Close()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func benchmarkPersisterOneKey(b *testing.B, key []byte, path string, id string) {
```
Comment (AdoAdoAdo) [inline leveldb/shardedLeveldbSerial_bench_test.go]: not used, can be removed

File: leveldb/shardedLeveldbSerial_bench_test.go [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+
+			<-maxRoutines
+			wg.Done()
+		}(key)
+	}
+
+	wg.Wait()
+}
+
+func getKey(
```
Comment (AdoAdoAdo) [inline leveldb/shardedLeveldbSerial_bench_test.go]: not used, can be removed

File: leveldb/shardedLeveldbSerial_bench_test.go [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+		value := generateRandomByteArray(32)
+
+		entries[string(key)] = value
+		keys = append(keys, string(key))
+	}
+
+	return entries, keys
+}
+
+func randInt(min int, max int) int {
```
Comment (AdoAdoAdo) [inline leveldb/shardedLeveldbSerial_bench_test.go]: not used, can be removed

File: leveldb/shardedLeveldbSerial_bench_test.go (hunk 3 above)
Comment (ssd04) [reply leveldb/shardedLeveldbSerial_bench_test.go]: ahh, right

File: leveldb/shardedLeveldbSerial_test.go (hunk 1 above)
Comment (ssd04) [reply leveldb/shardedLeveldbSerial_test.go]: done

File: leveldb/shardedLeveldbSerial_bench_test.go (hunk 2 above)
Comment (ssd04) [reply leveldb/shardedLeveldbSerial_bench_test.go]: removed, thanks

**Review by AdoAdoAdo — APPROVED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

**Pull request merged**